rdrf/rdrf/admin.py

```diff
@@ -130,104 +174,142 @@ def export_registry(registry, request):
-        response['Content-Disposition'] = 'attachment; filename="%s"' % name
+            zf.close()
+            zipped_file.flush()
+            zipped_file.seek(0)
 
-        return response
+            response = HttpResponse(FileWrapper(zipped_file), content_type='application/zip')
+            name = "export_" + export_time + "_" + \
+                   reduce(lambda x, y: x + '_and_' + y, [r.code for r in registries]) + ".zip"
+            response['Content-Disposition'] = 'attachment; filename="%s"' % name
 
+            return response
+    export_registry_action.short_description = _("Export")
 
-export_registry_action.short_description = "Export"
+    def generate_questionnaire_action(self, request, registry_models_selected):
+        for registry in registry_models_selected:
+            registry.generate_questionnaire()
 
+    generate_questionnaire_action.short_description = _("Generate Questionnaire")
 
-def design_registry_action(modeladmin, request, registry_models_selected):
-    if len(registry_models_selected) != 1:
-        return
-    else:
-        registry = [r for r in registry_models_selected][0]
-        return HttpResponseRedirect(reverse('rdrf_designer', args=(registry.pk,)))
+    def enable_registration_action(self, request, registry_models_selected):
+        unchanged = []
 
+        for registry in registry_models_selected:
+            if registry.has_feature(RegistryFeatures.REGISTRATION):
+                unchanged.append(registry.code)
 
-design_registry_action.short_description = _("Design")
+            registry.add_feature(RegistryFeatures.REGISTRATION)
+            registry.save()
 
+            messages.success(request, f"Registration enabled for '{registry.code}'")
 
-def generate_questionnaire_action(modeladmin, request, registry_models_selected):
-    for registry in registry_models_selected:
-        registry.generate_questionnaire()
+            existing_notifications = EmailNotification.objects.filter(
+                registry=registry,
+                description__in=EventType.REGISTRATION_TYPES
+            )
 
+            if len(existing_notifications) == 0:
+                messages.warning(request, f"Notifications need to be created for registry '{registry.code}'. "
+                                          f"Use the 'Create notifications' action")
 
-generate_questionnaire_action.short_description = _("Generate Questionnaire")
+        # Check for missing cache table
+        for cache_alias in settings.CACHES:
+            cache = caches[cache_alias]
+            if hasattr(cache, '_table') and cache._table not in connection.introspection.table_names():
+                messages.error(request, _(f"Missing cache table '{cache._table}'. Run django-admin createcachetable"))
 
+        if len(unchanged) > 0:
+            messages.info(request, f"'{', '.join(code for code in unchanged)}' already enabled registration")
```

Comment:
Any reason we're not adding default notifications for registration ourselves if they don't exist?

Reply:
OK. But I still don't like it that it is called "Create notifications". It isn't clear to the user what type of notifications we're creating, because there are many types of notifications.
Maybe call it something like "Create default registration notifications"?

Also, while I understand that you might want to avoid the addition of notifications the whole purpose of this feature is to make it easier to enable registration.
Perhaps we could add a button to the notification that calls the "Create default registration notifications" by default for the registries that don't have it.
So you would display the message warning the user the registries 'reg1, reg2' have registration enabled but don't have notifications configured. Would you like to <button>Create default registration notifications</button> or would you like to configure them yourself on the <a>Email Notifications</a> page?

The link would take you to the Email Notifications Admin page, the button would call the admin action that creates the default notifications just as the user selected all the registries that don't have it and selected and execute the Admin action manually.